src/Microsoft.AspNetCore.Dispatcher/Patterns/RoutePatternBuilder.cs

```diff
@@ -16,30 +15,36 @@ private RoutePatternBuilder()
 
         public IList<RoutePatternPathSegment> PathSegments { get; } = new List<RoutePatternPathSegment>();
 
-        public string Text { get; set; }
+        public string RawText { get; set; }
 
-        public RoutePatternBuilder AddPathSegment(RoutePatternPart part)
+        public RoutePatternBuilder AddPathSegment(params RoutePatternPart[] parts)
         {
-            return AddPathSegment(null, part, Array.Empty<RoutePatternPart>());
-        }
+            if (parts == null)
+            {
+                throw new ArgumentNullException(nameof(parts));
+            }
 
-        public RoutePatternBuilder AddPathSegment(RoutePatternPart part, params RoutePatternPart[] parts)
-        {
-            return AddPathSegment(null, part, Array.Empty<RoutePatternPart>());
-        }
+            if (parts.Length == 0)
+            {
+                throw new ArgumentException(Resources.RoutePatternBuilder_CollectionCannotBeEmpty, nameof(parts));
+            }
 
-        public RoutePatternBuilder AddPathSegment(string text, RoutePatternPart part)
-        {
-            return AddPathSegment(text, part, Array.Empty<RoutePatternPart>());
+            return AddPathSegment(null, parts);
         }
 
-        public RoutePatternBuilder AddPathSegment(string text, RoutePatternPart part, params RoutePatternPart[] parts)
```

Comment:
I decided to simplify this and just add an error message if you provide an empty list

Reply:
This API wasn't working well when I tried to consume it